File: SUMMER FS/5 may/program2.py
def getHint(secret, guess):
        bull, cow = 0, 0
        s = {} # secret hashtable
        g = {} # guess hashtable
        
        for i in range(len(secret)):
            if secret[i] == guess[i]:
                bull += 1
            else:
                s[secret[i]] = s.get(secret[i], 0) + 1
                g[guess[i]] = g.get(guess[i], 0) + 1
        
        for k in s:
            if k in g:
                cow += min(s[k], g[k])
        
        return '{0}B{1}C'.format(bull, cow)

secret=input()
guess=input()
print(getHint(secret,guess))

File: SUMMER FS/5 may/program2.java

import java.util.*;
class program2
{
	public static String getHint(String secret, String guess) {
    int bulls = 0;
    int cows = 0;
    int[] numbers = new int[10];
    for (int i = 0; i<secret.length(); i++) {
        if (secret.charAt(i) == guess.charAt(i)) bulls++;
        else {
            if (numbers[secret.charAt(i)-'0']++ < 0) cows++;
            if (numbers[guess.charAt(i)-'0']-- > 0) cows++;
        }
    }
    return bulls + "B" + cows + "C";
  }

	public static void main(String[] args) 
	{
		Scanner sc=new Scanner(System.in);
		String secret=sc.next();
		String guess=sc.next();
		System.out.println(getHint(secret,guess));
	}
}
